sdk/src/transaction_context.rs

```diff
@@ -615,9 +715,11 @@ impl<'a> BorrowedAccount<'a> {
     ///
     /// Fills it with zeros at the end if is extended or truncates at the end otherwise.
     pub fn set_data_length(&mut self, new_length: usize) -> Result<(), InstructionError> {
+        self.is_data_mutable(new_length)?;
         let mut total_resize_delta = self.transaction_context.total_resize_delta.borrow_mut();
         *total_resize_delta = total_resize_delta
             .saturating_add((new_length as i64).saturating_sub(self.get_data().len() as i64));
+        self.touch()?;
```

Comment:
why this `.touch()` isn't immediately following the `is_data_mutable()` calls like all other call-sites?

I think consistency is kind of necessity rather than nicety, considering the criticality of these code. :)